Source/FakeItEasy.Specs/DisposableSpecs.cs

```diff
@@ -0,0 +1,55 @@
+﻿namespace FakeItEasy.Specs
+{
+    using System;
+    using FluentAssertions;
+    using Machine.Specifications;
+
+    public class when_faking_a_disposable_class
+        : EventRaisingSpecs
+    {
+        static IDisposable fake;
+        private static Exception exception;
+
+        Establish context = () =>
+        {
+            AppDomain.CurrentDomain.UnhandledException += UnhandledExceptionHandler;
+
+            fake = A.Fake<SomeDisposable>();
+        };
+
+        Because of = () =>
+        {
+            fake = null;
+            GC.Collect();
+            GC.WaitForPendingFinalizers();
+        };

```

Comment:
I took it for a whirl locally and it does indeed fail (not that I doubted you, I just wanted to see it with my own eyes :wink:). I suppose, in this case, it's impossible to write a test which is *guaranteed* to fail, only one which *happens* to fail :confused:.